### PR TITLE
Add OTP 2 deployment enhancements (OTP 2 checkbox and custom files)

### LINKED ISSUE #519: Remove references to R5
## Observed behavior

Deployment in Data Tools offers an R5 option for a tool no longer used for OTP.

## Expected behavior

Option, and references in code, should be removed.

### PR DESCRIPTION
**Pull request by evansiroky**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [ ] All tests and CI builds passing

### Description

This adds <strike>a checkbox</strike> a dropdown to the DeploymentViewer component so that a user can select whether a jar file is an OTP 2.x jar. 

Also, it is now possible to add custom files to a deployment. These show up beneath the build and router config items.

This works in conjunction with https://github.com/ibi-group/datatools-server/pull/363.

This PR also removes references to r5 (fixes #519)

### CODE REVIEW

**Review by binh-dam-ibigroup — APPROVED** [review]
I think this is good for the time being.

**Comment by evansiroky** [comment]
Assigning back to myself to make deployments be able to include extra files (otp-config.json and walk-config.json).

**Comment by evansiroky** [comment]
Ready for a review again with custom file support for deployments.

File: lib/manager/components/DeploymentViewer.js
Comment (landonreed) [inline lib/manager/components/DeploymentViewer.js:952]: I think it's safe to say that this file is getting far too large! Could you please split it up a bit?

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Please address removing blank custom file entries that the user does not save, and the X in the OTP version dropdowns.

**Review by landonreed — CHANGES_REQUESTED** [review]
@evansiroky, I have more to look at, but before you do any refactoring on the custom file stuff, I think we need to come up with a more standard interface for setting these files that's shared between build/router config and the custom files. Perhaps I'll suggest some changes in a PR.

**Comment by landonreed** [comment]
Marking as Blocked until #641 is merged.

**Comment by evansiroky** [comment]
I added some more validation checks for custom files.

File: lib/manager/components/DeploymentViewer.js
Comment (binh-dam-ibigroup) [reply lib/manager/components/DeploymentViewer.js]: I'll give a pass on this one, I understand now why`<Select>` was preferred over `<FomrControl>`.

File: lib/manager/components/deployment/CustomFileEditor.js
Comment (binh-dam-ibigroup) [inline lib/manager/components/deployment/CustomFileEditor.js:176]: See my other comment regarding hiding Cancel when creating a new entry.

File: lib/manager/components/deployment/DeploymentConfigurationsPanel.js
Comment (binh-dam-ibigroup) [inline lib/manager/components/deployment/DeploymentConfigurationsPanel.js:161]: Is this always true?

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM. Please consider the minor suggestions regarding hiding the Cancel button when creating a new custom file and one maybe obsolete variable.

File: lib/manager/components/deployment/CustomFileEditor.js
Comment (landonreed) [inline lib/manager/components/deployment/CustomFileEditor.js:58]: Here and above in `_fileNameValid` I would suggest using `Boolean(customFile.contents)` rather than `!!customFile.contents`. I just find it infinitely more readable.

File: lib/manager/components/deployment/CustomFileEditor.js
Comment (landonreed) [inline lib/manager/components/deployment/CustomFileEditor.js:50]: Does this check for URI belong here? It seems like a simpler approach would just be:
```suggestion
    const {contents, filename} = customFile
    return fileSource === 'raw' && Boolean(contents) && Boolean(filename)
```

**Review by landonreed — APPROVED** [review]
Looks good, just a couple of suggestions.

File: lib/manager/components/deployment/CustomFileEditor.js
Comment (evansiroky) [reply lib/manager/components/deployment/CustomFileEditor.js:50]: Yes, it's required. The proposed change would return `false` for any custom file with a URI which is not correct since a filename is optional when a URI is provided.

Comment (evansiroky) [reply lib/manager/components/deployment/CustomFileEditor.js]: After looking at this more closely I ended up rewriting this so that it always returns `true` for a custom file with a URI. See 00445ee33e7b02c77a650d4a4098e64b6c528c67

File: lib/manager/components/deployment/CustomFileEditor.js
Comment (evansiroky) [reply lib/manager/components/deployment/CustomFileEditor.js]: I don't think `!!` is unreadable, but whatever, I made the change in 00445ee33e7b02c77a650d4a4098e64b6c528c67

File: lib/manager/components/deployment/DeploymentConfigurationsPanel.js
Comment (evansiroky) [reply lib/manager/components/deployment/DeploymentConfigurationsPanel.js:171]: For now, yes

File: lib/manager/components/deployment/CustomFileEditor.js
Comment (evansiroky) [reply lib/manager/components/deployment/CustomFileEditor.js:140]: Cancel button hiding logic implemented in 00445ee33e7b02c77a650d4a4098e64b6c528c67

File: lib/manager/components/deployment/CustomFileEditor.js
Comment (evansiroky) [reply lib/manager/components/deployment/CustomFileEditor.js]: Cancel button hiding logic implemented in 00445ee

**Comment by evansiroky** [comment]
Assigning back for one last review.

File: lib/manager/components/deployment/DeploymentConfigurationsPanel.js
Comment (binh-dam-ibigroup) [inline lib/manager/components/deployment/DeploymentConfigurationsPanel.js:38]: Would a timestamp or a uuid better than the file contents for identification purposes? (If it's a uuid, we won't need the other attributes to generate a unique key)

**Review by binh-dam-ibigroup — APPROVED** [review]
Please see the incidental comments. Also, please add in the PR description that it fixes #519.

File: lib/manager/components/deployment/DeploymentConfigurationsPanel.js
Comment (evansiroky) [reply lib/manager/components/deployment/DeploymentConfigurationsPanel.js:38]: This is the tricky thing about the custom files since they're saved as an array within the deployment object. Perhaps they should be in their own table and then they would have IDs. But since they don't, I think it makes sense to use these items to create a composite key.

File: lib/manager/components/deployment/DeploymentConfigurationsPanel.js
Comment (landonreed) [inline lib/manager/components/deployment/DeploymentConfigurationsPanel.js:41]: This formatting is kind of painful to me. OK if we "one-line" this if we just destructure the various `customFile` fields?

```suggestion
  const {contents, filename, uri} = customFile
  return `${filename || ''}-${contents || ''}-${uri || ''}-${idx}`
```

**Review by landonreed — APPROVED** [review]
Looks good, just one minor comment/suggestion.

**Pull request merged**
